src/core/net/srp_server.cpp

```diff
@@ -86,6 +87,7 @@ Server::Server(Instance &aInstance)
     , mMaxKeyLease(kDefaultMaxKeyLease)
     , mLeaseTimer(aInstance, HandleLeaseTimer)
     , mOutstandingUpdatesTimer(aInstance, HandleOutstandingUpdatesTimer)
+    , mServiceUpdateId(Random::NonCrypto::GetUint32())
```

Comment:
Use random initial value so that there will be small chances for IDs between resets of openthread stack to conflict with each other. This is for cases that the Advertising Proxy implementation may not be reset when openthread stack is reset.